src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -400,6 +414,98 @@ static void filter_bogus_addresses(struct in_addr *addresses, size_t *n) {
         *n = j;
 }
 
+/* Parses compressed domain names. */
+static int lease_parse_search_domains(const uint8_t *option, size_t len, char ***domains, size_t *domains_count) {
+        _cleanup_strv_free_ char **names = NULL;
+        size_t pos = 0, cnt = 0;
+        int r;
+
+        assert(domains);
+        assert(domains_count);
+        assert_return(option && len > 0, -ENODATA);
+
+        while (pos < len) {
+                _cleanup_free_ char *ret = NULL;
+                size_t n = 0, allocated = 0;
+                size_t jump_barrier = pos, next_chunk = 0;
+                bool first = true;
+
+                for (;;) {
+                        uint8_t c;
+                        c = option[pos++];
+
+                        if (c == 0) {
+                                /* End of name */
+                                break;
+                        } else if (c <= 63) {
+                                const char *label;
+
+                                /* Literal label */
+                                label = (const char *)&option[pos];
```

Comment:
appears like a strange way to write this:

```c
label = (const char*) (options + pos);
```

i.e. first dereffing via `[]`, and then taking the address of it via `&` sounds needlessly indirect...